### PR TITLE
PR for openapi-generator-maven-plugin inputSpec -- Allow jar: URLs

### PR DESCRIPTION
**Pull request by parenko**
This is a PR in response to #10016. It changes `<inputSpec/>` processing to:

1. Search a classpath of Maven compilation dependencies for resources.
2. Allow URLs of the form jar:jar-file-specific-url!/spec.yaml

In the case that a compilation classpath resource is specified, the resource URL is passed to the swagger-parser OpenAPIV3Parser instead of the `<inputSpec/>` string since the OpenAPIV3Parser does not have access to the Maven compilation classpath. This requires https://github.com/swagger-api/swagger-parser/issues/1592 and https://github.com/swagger-api/swagger-parser/pull/1593.

modules/openapi-generator-maven-plugin/src/test/java/org/openapitools/codegen/plugin/CodeGenMojoTest.java has been expanded to include tests for resource and URL input specs.

(This is a copy of PR #10037. I've rebased original branch on master, fixed merge conflicts and removed swagger-parser 2.0.28-SNAPSHOT since [2.1.19](https://github.com/OpenAPITools/openapi-generator/blob/master/pom.xml#L1255) is used now)

CC @wing328, @jimschubert

### CODE REVIEW

**Comment by wing328** [comment]
can you please also add a test or 2 in https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/maven-plugin-tests.yaml#L44 ?

**Comment by parenko** [comment]
> can you please also add a test or 2 in https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/maven-plugin-tests.yaml#L44 ?

Will do so. I have to fix other tests, also. I have so issues to run the build locally.

**Comment by parenko** [comment]
@wing328 instead of adding another pom to  https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/maven-plugin-tests.yaml#L44 , I've extended https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator-maven-plugin/examples/java-client.xml with another `execution`, pointing to a jar as `inputSpec`.

There are also tests in `CodeGenMojoTest.java`. For `inputSpec` as classpath, url or file there are other examples and tests, thus I did not additional examples.

File: modules/openapi-generator-maven-plugin/pom.xml [hunk 1]
```diff
@@ -64,6 +64,12 @@
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-resolver-provider</artifactId>
+            <version>3.9.4</version>
```
Comment (wing328) [inline modules/openapi-generator-maven-plugin/pom.xml:70]: can w use 3.9.6 instead which is the latest stable 3.x version?

Comment (parenko) [reply modules/openapi-generator-maven-plugin/pom.xml:70]: I've aligned it with other maven dependencies. Should I update other references to 3.9.6, also? Or just use a property to reference same maven version and update in another PR?

https://github.com/apache/maven/blob/master/pom.xml#L308

Comment (parenko) [reply modules/openapi-generator-maven-plugin/pom.xml]: Updated maven dependencies to 3.0.6 via same property. Now it is clearer that they belong together updated.

**Comment by wing328** [comment]
@parenko thanks for the PR, which has been merged into the master.

**Pull request merged**